spec/datadog/appsec/configuration/settings_spec.rb

```diff
@@ -0,0 +1,192 @@
+# typed: false
+
+require 'datadog/appsec/spec_helper'
+
+RSpec.describe Datadog::AppSec::Configuration::Settings do
+  shared_context 'registry with integration' do
+    let(:registry) { {} }
+    let(:integration_name) { :example }
+    let(:integration_options) { double('integration integration_options') }
+    let(:integration_class) { double('integration class', loaded?: false) }
+    let(:integration) do
+      instance_double(
+        Datadog::AppSec::Contrib::Integration::RegisteredIntegration,
+        klass: integration_class,
+        options: integration_options
+      )
+    end
+
+    before do
+      registry[integration_name] = integration
+
+      allow(Datadog::AppSec::Contrib::Integration).to receive(:registry).and_return(registry)
+    end
+  end
+
+  describe Datadog::AppSec::Configuration::Settings do
+    include_context 'registry with integration'
+
+    subject(:settings) { described_class.new }
+
+    let(:dsl) { Datadog::AppSec::Configuration::DSL.new }
+
+    after { settings.send(:reset!) }
+
+    describe '#enabled' do
+      subject(:enabled) { settings.enabled }
+      it { is_expected.to eq(true) }
+    end
+
+    describe '#enabled=' do
+      subject(:enabled_) { settings.merge(dsl.tap { |c| c.enabled = false }) }
+      it { expect { enabled_ }.to change { settings.enabled }.from(true).to(false) }
+    end
+
+    describe '#ruleset' do
+      subject(:ruleset) { settings.ruleset }
+      it { is_expected.to eq(:recommended) }
+    end
+
+    describe '#ruleset=' do
+      subject(:ruleset_) { settings.merge(dsl.tap { |c| c.ruleset = :risky }) }
+      it { expect { ruleset_ }.to change { settings.ruleset }.from(:recommended).to(:risky) }
+    end
+
+    describe '#waf_timeout' do
+      subject(:waf_timeout) { settings.waf_timeout }
+      it { is_expected.to eq(5000) }
+    end
+
+    describe '#waf_timeout=' do
+      subject(:waf_timeout_) { settings.merge(dsl.tap { |c| c.waf_timeout = 3 }) }
+      it { expect { waf_timeout_ }.to change { settings.waf_timeout }.from(5000).to(3) }
+    end
+
+    describe '#waf_debug' do
+      subject(:waf_debug) { settings.waf_debug }
+      it { is_expected.to eq(false) }
+    end
+
+    describe '#waf_debug=' do
+      subject(:waf_debug_) { settings.merge(dsl.tap { |c| c.waf_debug = true }) }
+      it { expect { waf_debug_ }.to change { settings.waf_debug }.from(false).to(true) }
+    end
+
+    describe '#trace_rate_limit' do
+      subject(:trace_rate_limit) { settings.trace_rate_limit }
+      it { is_expected.to eq(100) }
+    end
+
+    describe '#trace_rate_limit=' do
+      subject(:trace_rate_limit_) { settings.merge(dsl.tap { |c| c.trace_rate_limit = 2 }) }
+      it { expect { trace_rate_limit_ }.to change { settings.trace_rate_limit }.from(100).to(2) }
+    end
```

Comment:
It's actually `trace_rate_limit_per_second_per_concurrency_unit` (or `_per_thread`, or `_per_fiber` since it lives in a fiber local). Again I'd rather have proper docstrings explaining the unit+behaviour of the feature.